tools/build_api.py

```diff
@@ -312,9 +313,14 @@ def prepare_toolchain(src_paths, target, toolchain_name,
 
     # Toolchain instance
     try:
-        toolchain = TOOLCHAIN_CLASSES[toolchain_name](
-            target, options, notify, macros, silent,
-            extra_verbose=extra_verbose)
+        if toolchain_name == "GCC_ARM":
```

Comment:
I too felt it was weird. But that is because of the design. All three toolchains are expected get same no. of parameters? Perhaps toolchain constructors should accept keyword args so this check wasn't needed. 
Also this feature is specific to GCC only so this option should not be accepted for other toolchains.